src/main/java/org/apache/sling/caconfig/impl/ConfigurationResolverImpl.java

```diff
@@ -83,15 +91,54 @@ private void activate(Config config) {
 
     @Override
     public @NotNull ConfigurationBuilder get(@NotNull Resource resource) {
-        return new ConfigurationBuilderImpl(resource, this,
-                configurationResourceResolvingStrategy, configurationPersistenceStrategy,
-                configurationInheritanceStrategy, configurationOverrideMultiplexer, configurationMetadataProvider,
-                configBucketNames);
+        if (resource == null) {
+            return buildFor(resource);
```

Comment:
i do not understand this line - it is dead code assuming resource is never null according to @NotNull annotation